### ISSUE TITLE
Does this work with PowerShell?

### ISSUE THREAD

**Issue by kiedtl** [pawamoy/shell-history#27]
Will this work with PowerShell on Linux/Windows Powershell?

**Comment by pawamoy** [comment]
Not at all! But provided Powershell has ways of executing code before and after each interactive command, and disposes of a history which can be queried, it shouldn't be too hard to add support for it!

The shell code is only around 200 lines of code :slightly_smiling_face: 

I'm gonna see what Powershell has to offer (or if you already have some knowledge about it, don't hesitate to share here!)

**Comment by pawamoy** [comment]
Question on StackOverflow: https://stackoverflow.com/questions/55972059/powershell-execute-things-before-and-or-after-an-interactive-command

- [ ] before
- [x] after: using `prompt` function

I don't know Powershell at all so if anyone could look into the suggestions made in the comments it would be great!

**Comment by kiedtl** [comment]
I'm not sure about the internals of shellhistory, but wouldn't it be possible to do with only an `after` function? Or would some features be broken?

**Comment by pawamoy** [comment]
It would not be broken, but the generated data could be inaccurate. If a command changes the working directory, or overrides itself or deletes itself (think of virtualenvs' `deactivate` function: it unsets itself at the end), then we could not compute its true type (file/function/alias etc).

We also need to get a timestamp before the command, to be able to compute its duration.

**Comment by pawamoy** [comment]
However, maybe there's clever ways to do that in Powershell! Since it's a completely different shell, Bash/Zsh logic might not apply.
